### PR TITLE
AdoptAWidget: MaterialBanner

### LINKED ISSUE #1234: There's no way to follow a widget's position
Suppose you have a widget that's in a list, inside a layout-based transition, inside an animating transform, etc, and you want to overlay something above it, for example a tooltip. There's no way to do this today. None of the widgets notify anyone when they decide to scroll or reposition children or transform them in paint (or in the compositor!) or whatever.

### LINKED ISSUE #69474: Adopt a Widget - MaterialBanner
The [MaterialBanner](https://master-api.flutter.dev/flutter/material/MaterialBanner-class.html) widget needs:

- A code / DartPad sample

For details about how to participate, see [flutter.dev/adoptawidget](https://flutter.dev/adoptawidget)

### PR DESCRIPTION
**Pull request by tadaspetra**
AdoptAWidget: MaterialBanner

This pull request is:

- [x] A code sample
- [ ] More references
- [ ] More explanation

closes #69474

### CODE REVIEW

**Comment by tadaspetra** [comment]
> Thanks for your pull request. It looks like this may be your first contribution to a Google open source project (if not, look below for help). Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).
> 
> 📝 **Please visit https://cla.developers.google.com/ to sign.**
> 
> Once you've signed (or fixed any issues), please reply here with `@googlebot I signed it!` and we'll verify it.
> 
> #### What to do if you already signed the CLA
> ##### Individual signers
> * It's possible we don't have your GitHub username or you're using a different email address on your commit. Check [your existing CLA data](https://cla.developers.google.com/clas) and verify that your [email is set on your git commits](https://help.github.com/articles/setting-your-email-in-git/).
> 
> ##### Corporate signers
> * Your company has a Point of Contact who decides which employees are authorized to participate. Ask your POC to be added to the group of authorized contributors. If you don't know who your Point of Contact is, direct the Google project maintainer to [go/cla#troubleshoot](http://go/cla#troubleshoot) ([Public version](https://opensource.google/docs/cla/#troubleshoot)).
> * The email used to register you as an authorized contributor must be the email used for the Git commit. Check [your existing CLA data](https://cla.developers.google.com/clas) and verify that your [email is set on your git commits](https://help.github.com/articles/setting-your-email-in-git/).
> * The email used to register you as an authorized contributor must also be [attached to your GitHub account](https://github.com/settings/emails).
> 
> ℹ️ **Googlers: [Go here](https://goto.google.com/prinfo/https%3A%2F%2Fgithub.com%2Fflutter%2Fflutter%2Fpull%2F69530) for more info**.

@googlebot I signed it!

**Review by shihaohong — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/material/banner.dart [hunk 1]
```diff
@@ -18,6 +18,30 @@ import 'theme.dart';
 /// They are persistent and non-modal, allowing the user to either ignore them or
 /// interact with them at any time.
 ///
+/// {@tool dartpad --template=stateless_widget_scaffold}
+///
+/// ```dart
+/// Widget build(BuildContext context) {
+///   return MaterialBanner(
+///     padding: const EdgeInsets.all(20),
+///     content: Text("Hey, I am an Material Banner"),
```
Comment (Abhishek01039) [inline packages/flutter/lib/src/material/banner.dart:27]: `I am an Material Banner` here is a grammatical mistake.
Here should be `I am a Material Banner`
Thanks.

Comment (johnpryan) [reply packages/flutter/lib/src/material/banner.dart:27]: @Abhishek01039 good catch - do you want to fix this in a separate PR?

Comment (Abhishek01039) [reply packages/flutter/lib/src/material/banner.dart:27]: Yes.

**Comment by johnpryan** [comment]
@tadaspetra Thanks for participating in AdoptAWidget. We would like to send you a special thanks. If you are interested, please send an email to nyener at google dot com and provide a link to this issue.

**Comment by tadaspetra** [comment]
> @tadaspetra Thanks for participating in AdoptAWidget. We would like to send you a special thanks. If you are interested, please send an email to nyener at google dot com and provide a link to this issue.

I didn't get any response to the email 😕

**Comment by johnpryan** [comment]
It will likely take a few extra days, since we are on holiday in the U.S. Thanks for your patience!

**Comment by tadaspetra** [comment]
> It will likely take a few extra days, since we are on holiday in the U.S. Thanks for your patience!

ah no worries! I was just checking in to make sure, i didn't send the wrong email or something 😊

**Pull request merged**